manifests/server/config/zookeeper.pp

```diff
@@ -0,0 +1,77 @@
+# clickhouse::server::config::zookeeper
+# Set proper zookeeper config
+#
+# @summary Set proper zookeeper config
+#
+# @param nodes
+#   hash of `zk_server`: `id` to define nodes
+#
+# @param port
+#   cilent port of zookeeper cluster
+#
+# @param session_timeout_ms
+#   maximum timeout for client session in milliseconds
+#
+# @param operation_timeout_ms
+#   maximum timeout for operation in milliseconds
+#
+# @param root
+#   ZNode, that is used as root for znodes used by ClickHouse server
+#
+# @param user
+#   user if zookeeper uses authorization
+#
+# @param password
+#   password if zookeeper uses authorization
+#
+# @param service_notify
+#   If ClickHouse server should be restarted on the config update
+#
+# @example Use with params
+#   class { 'clickhouse::server::config::zookeeper':
+#       nodes    => { 'server1' => 1},
+#       user     => 'user',
+#       password => 'password',
+#   }
+#
+# @author InnoGames GmbH
+#
+class clickhouse::server::config::zookeeper (
+    Hash[String[1], Integer[1, 255]] $nodes,
+    Integer[1, 65536]                $port                 = 2181,
+    Integer[1]                       $session_timeout_ms   = 30000,
+    Integer[1]                       $operation_timeout_ms = 10000,
+    Optional[Stdlib::Unixpath]       $root                 = undef,
+    Optional[String[1]]              $user                 = undef,
+    Optional[String[1]]              $password             = undef,
```

Comment:
Does it make sense to enforce at lease a minimum length for the password ?

Reply:
About the minimum length, I am not sure. This one is out of the CH scope, so ZK admin should decide. The `String[1]` is the least conflicting option here, IMHO.